src/main/java/org/opentripplanner/transit/raptor/rangeraptor/RangeRaptorWorker.java

```diff
@@ -239,30 +246,33 @@ private void findAllTransitForRound() {
                     if(pattern.boardingPossibleAt(stopPos)) {
                         // MC Raptor have many, while RR have one boarding
                         transitWorker.forEachBoarding(stopIndex, (int prevArrivalTime) -> {
-                            RaptorTripScheduleBoardOrAlightEvent<T> result = null;
-
-                            if(enableGuaranteedTransfers) {
-                                // Board using guaranteed transfers
-                                result = findGuaranteedTransfer(
-                                        route.timetable(),
-                                        txService, stopIndex, stopPos
-                                );
-                            }
-
-                            // Find the best trip and board [no guaranteed transfer exist]
-                            if(result == null) {
-                                this.earliestBoardTime = earliestBoardTime(prevArrivalTime);
-                                // check if we can back up to an earlier trip due to this stop
-                                // being reached earlier
-                                result = tripSearch.search(
-                                        earliestBoardTime,
-                                        stopPos,
-                                        transitWorker.onTripIndex()
-                                );
-                            }
-
-                            if (result != null) {
-                                transitWorker.board(stopIndex, earliestBoardTime, result);
+                            // skip stop if its a forbidden stop to stop transfers
+                            if(!enableForbiddenTransfers || !isForbiddenTransfer(txForbiddenService, stopIndex, stopPos)) {
+                                RaptorTripScheduleBoardOrAlightEvent<T> result = null;
+
+                                if(enableGuaranteedTransfers) {
+                                    // Board using guaranteed transfers
+                                    result = findGuaranteedTransfer(
+                                            route.timetable(),
+                                            txService, stopIndex, stopPos
+                                    );
+                                }
```

Comment:
I do not want every small feature to add its own special code here, I want something generic that can be used to solve several use-cases. But, if you get this to work, it will be a good starting point and we can make it generic then.

Reply:
Ah, it was not clear for me that GuaranteedTransfers was supposed to be generic.